client_plugin/drivers/shared/shared_driver.go

```diff
@@ -111,7 +233,118 @@ func (d *VolumeDriver) UnmountVolume(name string) error {
 
 // Create - create a volume.
 func (d *VolumeDriver) Create(r volume.Request) volume.Response {
-	log.Errorf("VolumeDriver Create to be implemented")
+	log.Infof("Entering VolumeDriver Create")
+	var msg string
+	var hnames []string
+	var entries []kvPair
+
+	// Ensure that the node running this command is a manager
+	info, err := d.dockerd.Info(context.Background())
+	if err != nil {
+		msg = fmt.Sprintf("Failed to get Info from docker client. Reason: %v", err)
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+	if info.Swarm.ControlAvailable == false {
+		msg = fmt.Sprintf("This node is not a swarm manager.")
+		msg += fmt.Sprintf(" Shared Volume creation is only possible from swarm manager nodes.")
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+
+	// Initialize volume metadata in KV store
+	hostname, err := os.Hostname()
+	if err != nil {
+		msg = fmt.Sprintf("Failed to fetch hostname during volume creation")
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+	hnames = append(hnames, hostname)
+	volRecord := VolumeMetadata{
+		Status:         volStateCreating,
+		GlobalRefcount: 0,
+		Port:           0,
+		ServiceName:    "Not started yet",
+		Username:       sambaUsername,
+		Password:       sambaPassword,
+		ClientList:     hnames,
+	}
+
+	// Append global refcount and status to kv pairs that will be written
+	entries = append(entries, kvPair{key: etcdPrefixGRef + r.Name, value: strconv.Itoa(volRecord.GlobalRefcount)})
+	entries = append(entries, kvPair{key: etcdPrefixState + r.Name, value: string(volRecord.Status)})
+	// Append the rest of the metadata as one KV pair where the data is jsonified
+	byteRecord, err := json.Marshal(volRecord)
+	if err != nil {
+		msg = fmt.Sprintf("Cannot create volume. Failed to marshal metadata to json. Reason: %v", err)
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+	entries = append(entries, kvPair{key: etcdPrefixInfo + r.Name, value: string(byteRecord)})
+
+	log.Infof("Attempting to write initial metadata entry for %s", r.Name)
+	err = d.writeVolMetadata(entries)
+	if err != nil {
+		msg = fmt.Sprintf("Failed to create volume %s. Reason: %v",
+			r.Name, err)
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+
+	// Create traditional volume as backend to shared volume
+	msg = fmt.Sprintf("Attempting to create internal volume for %s", r.Name)
+	log.Infof(msg)
+	internalVolname := tradVolumePrefix + r.Name
+	dockerVolOptions := dockerTypes.VolumeCreateRequest{
+		Driver:     d.internalVolumeDriver,
+		Name:       internalVolname,
+		DriverOpts: r.Options,
+	}
+	_, err = d.dockerd.VolumeCreate(context.Background(), dockerVolOptions)
```

Comment:
Ok, if the host crashes before or after volumeCreate by the backing volume driver then on startup this plugin must go thru and figure which volumes are in creating state and remove those or update to _ready_ state if a backing volume is successfully created.